prov/shm/src/smr.h

```diff
@@ -171,14 +171,15 @@ static inline int smr_match_tag(uint64_t tag, uint64_t ignore, uint64_t match_ta
 }
 
 static inline enum fi_hmem_iface smr_get_mr_hmem_iface(struct util_domain *domain,
-				void **desc, uint64_t *device)
+				void **desc, uint64_t *device, uint64_t *flags)
```

Comment:
This function call now does more than its name suggests.  The flags can be read directly from the desc, and in half the use cases, the returned flags value is ignored.  It seems like it would be cleaner to have a macro to just return the mr flags and call it directly where needed.

Reply:
That definitely makes more sense. Updated.